### PR TITLE
feat(spanner): add instance autoscaling attributes to spanner instance resource

### PR DESCRIPTION
**Pull request by rahul2393**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
spanner: added `autoscaling_config` field to  `google_spanner_instance`
```

### CODE REVIEW

**Comment by rahul2393** [comment]
@zli82016 can you please help review this

**Comment by zli82016** [comment]
The API documentation is not updated to have the new field. Will the API doc be updated soon? 
https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances

File: mmv1/third_party/terraform/services/spanner/resource_spanner_instance_test.go [hunk 1]
```diff
@@ -142,6 +142,30 @@ func TestAccSpannerInstance_virtualUpdate(t *testing.T) {
...
+				Config: testAccSpannerInstance_basicWithAutoscalerConfigUsingProcessingUnitsAsConfigs(displayName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("google_spanner_instance.basic", "state"),
+				),
+			},
+			{
+				ResourceName:      "google_spanner_instance.basic",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/spanner/resource_spanner_instance_test.go:164]: Can you please add the update tests to test updating the new added fields to new value?

File: mmv1/products/spanner/Instance.yaml [hunk 2]
```diff
@@ -156,3 +158,50 @@ properties:
...
+      - autoscaling_config
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'autoscalingLimits'
+        description: |
+          Defines scale in controls to reduce the risk of response latency
+          and outages due to abrupt scale-in events
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'minProcessingUnits'
```
[the remark this answers is not in the document]
Comment (rahul2393) [reply mmv1/products/spanner/Instance.yaml:180]: In the API documentation side its mentioned that 
```
Maximum number of processing units allocated to the instance. If set,
this number should be multiples of 1000 and be greater than or equal to
min_processing_units
```
In client libraries also we don't set default values and let server handle the validation.

**Comment by rahul2393** [comment]
> The API documentation is not updated to have the new field. Will the API doc be updated soon?

Yes its scheduled to be announced in public next week, proto changes and client library changes has already been merged, we need Terraform support before public announcement.

**Comment by zli82016** [comment]
The test failed. It looks like the new field needs to be added to `fieldMask` in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/encoders/spanner_instance_update.go.erb

**Comment by rahul2393** [comment]
> The test failed. It looks like the new field needs to be added to `fieldMask` in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/encoders/spanner_instance_update.go.erb


Thanks @zli82016 for pointing me this, have incorporated the changes and added tests. please help review.
Screenshot 2023-11-08 at 00 35 12

**Comment by zli82016** [comment]
Can you please fix the unit tests? Thanks.

**Comment by rahul2393** [comment]
> Can you please fix the unit tests, @rahul2393 ? Thanks.

Fixed!!

**Review by zli82016 — APPROVED** [review]
Thanks for the fix. LGTM.

**Pull request merged**